src/langcheck/metrics/ja/pairwise_text_quality.py

```diff
@@ -0,0 +1,256 @@
+
+        深呼吸をして、この問題をステップバイステップで取り組んでください。
+        '''
+
+    def _prompt_with_source_and_reference(gen_output_a: str, gen_output_b: str,
+                                          user_query: str, ref_output: str,
+                                          source: str) -> str:
+        return f'''
+        ユーザーの質問に対する2つの回答の品質を比較してください。ユーザーの質問に関連があると
+        思われるソーステキストも参考として提供されます（ソーステキストには重複が含まれている
+        可能性があります）。さらに、ユーザーの質問に対する理想的な回答も参考として提供されます。
+        データは以下の通りです:
+        [BEGIN DATA]
+        ************
+        [ユーザーの質問]: {user_query}
+        ************
+        [ソース]: {source}
+        ************
+        [理想的な回答]: {ref_output}
+        ************
+        [Response A]: {gen_output_a}
+        ************
+        [Response B]: {gen_output_b}
+        ************
+        [END DATA]
+
+        ユーザーの質問に対してどちらの回答がより良いかを決定してください。提供されたソースと
+        理想的な回答を参考に、有用性、正確さ、関連性などの要素を考慮して評価してください。
+        回答が提示された順序に評価が影響されないようにしてください。回答の長さが評価に影響を
+        与えないようにしてください。利用可能な評価は以下の通りです:
+        `Response A` - Response Aがより良い回答です。
+        `Response B` - Response Bがより良い回答です。
+        `Tie` - 2つの回答は品質がほぼ同等です。
+
+        深呼吸をして、この問題をステップバイステップで取り組んでください。
+        '''
+
+    def _function_call_prompt(long_assessment: str) -> str:
+        return f'''
+        The following is an assessment on whether Response A or Response B is
+        the better response to the user's query:
+        ************
+        [Assessment]: {long_assessment}
+        ************
+
+        Save the resulting assessment. The available assessments are:
+        `Response A`
+        `Response B`
+        `Tie`
+        '''
+
+    pairwise_comparison_assessment_to_score = {
+        'Response B': 1.0,
+        'Tie': 0.5,
+        'Response A': 0.0
+    }
+    oai_evaluator = OpenAIBasedEvaluator(
+        assessment_to_score_mapping=pairwise_comparison_assessment_to_score,
+        function_name='save_pairwise_comparison_assessment',
+        function_description=("Saves a pairwise comparison assessment."),
+        argument_name='pairwise_comparison',
+        argument_description='The pairwise comparison assessment',
+        client_type=model_type,
+        client=openai_client,
+        openai_args=openai_args,
+        use_async=use_async)
+
+    prompt_generator = PairwiseComparisonPromptGenerator(
+        _prompt, _prompt_with_reference, _prompt_with_source,
+        _prompt_with_source_and_reference)
+    all_prompts = prompt_generator.generate_prompts(generated_outputs_a,
+                                                    generated_outputs_b,
+                                                    prompts, sources_a,
+                                                    sources_b,
+                                                    reference_outputs)
+    scores, explanations = oai_evaluator.get_score(all_prompts,
+                                                   _function_call_prompt)
+
+    if enforce_consistency:
```

Comment:
We might want to show something to the user before moving into the consistency check. When I see the progress bars twice in for the first time I was a bit confused.

Reply:
I made it so that the tqdm descriptions can be overridden, so now it looks like this. WDYT?
<img width="1882" alt="image" src="https://github.com/citadel-ai/langcheck/assets/107823399/a221194a-5d32-41d4-a3c7-4208e43e3e60">

(I tried just adding a print statement at first, but then it's a bit weird for async where the progress bars don't show up right now)

Reply:
Got it, it looks reasonable to me!!